enacts/onset/agronomy.py

```diff
@@ -128,43 +135,73 @@ def soil_plant_water_balance(
         * station        (station) int64 0 1
     """
     
-    # First Step
-    if np.size(et) == 1:
-        et = xr.DataArray(et)
-    # Setting Kc
+    # Initializations
+    et = xr.DataArray(et)
     if kc_params is None:
         kc = 1
-    else:
+        et_crop = et
+    else: #et_crop depends on et, kc and planting_date dims, and time_dim
         kc_inflex = kc_params.assign_coords(
             kc_periods=kc_params["kc_periods"].cumsum(dim="kc_periods")
         )
-        if planting_date is not None:
+        if planting_date is not None: # distance between 1st and planting days
             planted_since = peffective[time_dim][0].drop_vars(time_dim) - planting_date
-            kc = kc_inflex.interp(
-                kc_periods=planted_since, kwargs={"fill_value": 1}
-            ).drop_vars("kc_periods")
-    # Initializations of sm, drainage and et_crop
-    et_crop0 = kc * et.isel({time_dim: 0}, missing_dims='ignore', drop=True)
-    sm0, drainage0 = soil_plant_water_step(
-        sminit,
-        peffective.isel({time_dim: 0}, drop=True),
-        et_crop0,
-        taw,
+        else: # 1st day is planting day if sminit met condition
+            if sm_threshold is not None:
+                planted_since = xr.where(
+                    sminit >= sm_threshold, 0, np.nan
+                ).astype("timedelta64[D]")
+            else:
+                raise Exception("if planting_date is None, then define a sm_threshold")
+        et_crop = (xr.full_like(et, fill_value=np.nan)
+            + xr.zeros_like(kc_params.isel({"kc_periods": 0}, drop=True))
+            + xr.zeros_like(peffective[time_dim], dtype=et.dtype)
+            + xr.zeros_like(planted_since, dtype=et.dtype)
```

Comment:
Yes. I am almost certain I could use broadcast. It was done so in some other version of this work. Note that broadcast also creates "views" so "copy" will be needed at some point.
I don't have a sense what's best. It seems more elegant to use broadcast rather than something else to get side effects only. But we should also do what's more efficient, we should be broadcast also, it seems.

Reply:
just pushed version using broadcast